apps/backend/src/modules/users/user.repository.ts

```diff
@@ -3,60 +3,175 @@ import { type UserPasswordDetails } from "~/modules/users/libs/types/types.js";
+			.leftJoin(
+				"groups_to_permissions",
+				"groups.id",
+				"groups_to_permissions.group_id",
+			)
+			.leftJoin(
+				"permissions",
+				"groups_to_permissions.permission_id",
+				"permissions.id",
+			)
+			.execute();
 
-		return users.map((user) => UserEntity.initialize(user));
+		return users.map((user) =>
+			UserEntity.initialize({
+				email: user.email,
+				firstName: user.firstName,
+				group: user.group ? GroupEntity.initialize(user.group) : null,
+				groupId: user.groupId,
+				id: user.id,
+				lastName: user.lastName,
+				passwordHash: user.passwordHash,
+				passwordSalt: user.passwordSalt,
+			}),
+		);
 	}
 
 	public async findByEmail(email: string): Promise<null | UserEntity> {
-		const user = await this.userModel.query().where("email", email).first();
+		const user = await this.userModel
+			.query()
+			.where("email", email)
+			.withGraphJoined("group.permissions")
+			.first();
 
-		return user ? UserEntity.initialize(user) : null;
+		if (!user) {
+			return null;
+		}
+
+		return UserEntity.initialize({
+			email: user.email,
+			firstName: user.firstName,
+			group: user.group ? GroupEntity.initialize(user.group) : null,
+			groupId: user.groupId,
+			id: user.id,
+			lastName: user.lastName,
+			passwordHash: user.passwordHash,
+			passwordSalt: user.passwordSalt,
+		});
 	}
 
 	public async findById(id: number): Promise<null | UserEntity> {
-		const user = await this.userModel.query().findById(id);
+		const user = await this.userModel
+			.query()
+			.findById(id)
+			.withGraphJoined("group.permissions")
+			.first();
 
-		return user ? UserEntity.initialize(user) : null;
+		if (!user) {
+			return null;
+		}
+
+		return UserEntity.initialize({
+			email: user.email,
+			firstName: user.firstName,
+			group: user.group ? GroupEntity.initialize(user.group) : null,
+			groupId: user.groupId,
+			id: user.id,
+			lastName: user.lastName,
+			passwordHash: user.passwordHash,
+			passwordSalt: user.passwordSalt,
+		});
 	}
 
 	public async findPasswordDetails(
 		email: string,
 	): Promise<null | UserPasswordDetails> {
 		const user = await this.userModel
 			.query()
-			.select("id", "firstName", "lastName", "passwordHash", "passwordSalt")
-			.where("email", email)
+			.select(
+				"users.id",
+				"users.first_name",
+				"users.last_name",
+				"users.password_hash as passwordHash",
+				"users.password_salt as passwordSalt",
+				"users.group_id as groupId",
+			)
+			.withGraphJoined("group.permissions")
+			.where("users.email", email)
 			.first();
 
-		return user ?? null;
+		if (!user || !user.group || !user.group.permissions) {
+			return null;
+		}
+
+		return {
+			firstName: user.firstName,
+			group: {
+				id: user.group.id,
+				key: user.group.key,
+				name: user.group.name,
+				permissions: user.group.permissions.map((permission) => ({
+					id: permission.id,
+					key: permission.key,
+					name: permission.name,
+				})),
+			},
```

Comment:
is there an issue with not using it the same way as in other methods? i'm ok with remapping overall, but just want it to be consistent
```suggestion
			group: GroupEntity.initialize(user.group).toObject()
```